common.gradle

```diff
@@ -8,7 +8,7 @@ apply plugin: 'maven'
 group = 'org.jmonkeyengine'
 version = jmePomVersion
 
-sourceCompatibility = '1.7'
+sourceCompatibility = '1.8'
```

Comment:
Keep in mind that Java 7 class files can work in both Java 7 and 8 (and higher) JREs.
If you bump to Java 8 you don't have that luxury.

Which milestone will this end up in? How strict is the JME team with respect to semver?

You can also keep the warning around, as long as you're careful it's no problem.
Then for the time being you don't have to drop Java 7 support.